include/openssl/err.h

```diff
@@ -163,12 +163,25 @@ struct err_state_st {
 #  define X509err(f, r) ERR_raise_data(ERR_LIB_X509, (r), NULL)
 # endif
 
+/*
+ * Error codes are packed into a single 32-bit number, and have three parts,
+ * library (L), function (F, not used as of 3.0), and reason (R). It is
+ * split as follows:
+ *      LL FFF RRR
+ * The top two bits of LL are zero and fatal-error-flag.
+ * This is part of the ABI, and therefore an implied part of the API.
```

Comment:
Actually, since 3.0 only needs to be source compatible, you don't need to worry about the ABI

Reply:
... in other word, it should be safe to simply ditch the function number altogether, and thereby make the packed error code change to this structure:

    LL RRRRRR

Reply:
Ditching the function number was more than I first wanted to do, but will do so if you think I should.  I would save it as ```FLLRRRRR``` where is a flags nibble, currently must be zero.